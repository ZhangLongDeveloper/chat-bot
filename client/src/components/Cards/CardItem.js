import React from 'react';
import PropTypes from 'prop-types';

const CardItem = ({ data }) => {
  const {
    title, image, description, link,
  } = data;

  const renderLink = () => {
    if (link) {
      return (
        <div className="card-action">
          <a href={link}>View</a>
        </div>
      );
    }
    return null;
  };

  return (
    <div className="card-containter">
      <div className="card">
        <div className="card-image">
          <img src={image} alt="card-pic" />
          <span className="card-title">{title}</span>
        </div>
        <div className="card-content">
          <p>{description}</p>
        </div>
        {renderLink()}
      </div>
    </div>
  );
};

CardItem.propTypes = {
  data: PropTypes.object.isRequired,
};

export default CardItem;
