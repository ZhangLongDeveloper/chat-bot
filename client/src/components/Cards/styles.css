.cards-containter {
  overflow: auto;
  overflow-y: scroll;
}

.cards-containter .card-containter {
  display: inline-block;
  width: 200px;
  margin: 0 10px;
}