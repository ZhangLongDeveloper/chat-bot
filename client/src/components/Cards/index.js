import React from 'react';
import PropTypes from 'prop-types';

import Card from './CardItem';

import './styles.css';

const Cards = ({ data }) => {
  if (data && data.value && data.value.length) {
    return (
      <div className="cards-containter">
        <div style={{ width: 220 * data.value.length }}>
          {data.value.map(item => <Card data={item} key={item.id} />)}
        </div>
      </div>
    );
  }
  return <span className="red-text">Oops, something went wrong... :(</span>;
};

Cards.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Cards;
