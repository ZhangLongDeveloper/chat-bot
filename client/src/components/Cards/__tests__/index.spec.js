import React from 'react';
import { create } from 'react-test-renderer';

import Cards from '../../Cards';

jest.mock('../../Cards/CardItem', () => 'CardItem');

const mockComponent = props => (
  <Cards {...props} />
);

describe('components/<Cards />', () => {
  it('should render component with no data', () => {
    const tree = create(mockComponent({})).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with data', () => {
    const props = {
      data: {
        value: [
          { id: 'id-1' },
          { id: 'id-2' },
          { id: 'id-3c' },
        ],
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
