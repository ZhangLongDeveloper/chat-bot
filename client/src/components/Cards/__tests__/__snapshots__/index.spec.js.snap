// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/<Cards /> should render component with data 1`] = `
<div
  className="cards-containter"
>
  <div
    style={
      Object {
        "width": 660,
      }
    }
  >
    <CardItem
      data={
        Object {
          "id": "id-1",
        }
      }
    />
    <CardItem
      data={
        Object {
          "id": "id-2",
        }
      }
    />
    <CardItem
      data={
        Object {
          "id": "id-3c",
        }
      }
    />
  </div>
</div>
`;

exports[`components/<Cards /> should render component with no data 1`] = `
<span
  className="red-text"
>
  Oops, something went wrong... :(
</span>
`;
